File: 04.Simulation模拟/leetcode_1758_minOperations.py
class Solution:
    def minOperations(self, s: str) -> int:
        res1 = 0
        cur = 0
        for i in s:
            if i == str(cur):
                res1 += 1
            cur ^= 1
        return min(res1, len(s)-res1)

if __name__ == "__main__":
    sl = Solution()
    s = "10"
    print(sl.minOperations(s))

File: 04.Simulation模拟/leetcode_1758_minOperations.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    int minOperations(string s) {
        int res1 = 0, res2 = 0;
        int cur = 0;
        for (auto& c : s){
            if (cur == (c-'0')){
                res2++;
            }
            else{
                res1++;
            }
            cur ^= 1;
        }
        return min(res1, res2);
    }
};

int main(){
    Solution sl;
    string s = "10";
    cout << sl.minOperations(s) << endl;
}
